src/tokens.js

```diff
@@ -59,6 +60,52 @@ export function unsetTokens() {
   unsetUser();
 }
 
+/**
+ * Client-side check:
+ * Determine whether the access token is present and unexpired
+ * @returns {Boolean}
+ */
+export function isAccessTokenLocallyValid() {
+  return isJwtLocallyValid(store.tokens.accessToken);
+}
+
+/**
+ * Client-side check:
+ * Determine whether the refresh token is present and unexpired
+ * @returns {Boolean}
+ */
+export function isRefreshTokenLocallyValid() {
+  return isJwtLocallyValid(store.tokens.refreshToken);
+}
+
+/**
+ * Determine whether a user is logged in by checking their
+ * JWT access token and, if invalid, refreshing it and checking
+ * again.
+ * @returns {Boolean}
```

Comment:
```suggestion
 * @returns {Promise<Boolean>}
```

(if this remains async)

Reply:
Updated, thanks!